### ISSUE TITLE
Unable to self-update through a https intercepting proxy

### ISSUE THREAD

**Issue by splinter98** [jdxcode/rtx#714]
**Describe the bug**
self-update fails when running behing a https intercepting proxy.
```
Checking target-arch... linux-x64
Checking current version... v1.34.1
Checking latest released version... rtx ReqwestError: error sending request for url (https://api.github.com/repos/jdxcode/rtx/releases/latest): error trying to connect: invalid peer certificate: UnknownIssuer
rtx Run with RTX_DEBUG=1 for more information
```

**To Reproduce**

Prerequisites:
  * Ensure a HTTPS intercepting proxy (commonly found in enterprise environments) is active
  * HTTPS Intercepting root CA certificate is installed into OS certificate store

Steps:
  1.  `curl -sL https://github.com/jdxcode/rtx/releases/download/v1.34.1/rtx-v1.34.1-linux-x64 > rtx`
  2.  `chmod +x rtx`
  3. `./rtx self-update`

Output:
```shell
$./rtx self-update
Checking target-arch... linux-x64
Checking current version... v1.34.1
Checking latest released version... rtx ReqwestError: error sending request for url (https://api.github.com/repos/jdxcode/rtx/releases/latest): error trying to connect: invalid peer certificate: UnknownIssuer
rtx Run with RTX_DEBUG=1 for more information
```
**Expected behavior**
rtx does not error and can contact github to be able to check and download the latest version by using the system installed certificate

**`rtx doctor` output**
```
./rtx doctor
rtx version:
  1.34.1 linux-x64 (76f8de3 2023-07-14)

build:
  Target: x86_64-unknown-linux-gnu
  Features: SELF_UPDATE
  Built: Fri, 14 Jul 2023 17:59:02 +0000
  Rust Version: rustc 1.70.0 (90c541806 2023-05-31)
  Profile: release

shell:
  /bin/bash
  GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)
  Copyright (C) 2020 Free Software Foundation, Inc.
  License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

  This is free software; you are free to change and redistribute it.
  There is NO WARRANTY, to the extent permitted by law.

rtx environment variables:
  RTX_SHELL=bash

settings:
  {"always_keep_download": "false", "always_keep_install": "false", "asdf_compat": "false", "disable_default_shorthands": "false", "disable_tools": "[]", "experimental": "true", "jobs": "4", "legacy_version_file": "true", "legacy_version_file_disable_tools": "[]", "log_level": "INFO", "missing_runtime_behavior": "warn", "plugin_autoupdate_last_check_duration": "10080", "raw": "false", "trusted_config_paths": "[]", "verbose": "false"}

config files:
  /home/stephen/.config/rtx/config.toml
  /home/stephen/.tool-versions

plugins:
  bitwarden    https://github.com/michaeldelago/asdf-bitwarden#d869c04
  chezmoi      https://github.com/joke/asdf-chezmoi.git#07ddeac
  cmake        https://github.com/asdf-community/asdf-cmake#d631481
  github-cli   https://github.com/bartlomiejdanek/asdf-github-cli.git#9e0a9d7
  go           (core)
  java         (core)
  jq           https://github.com/azmcode/asdf-jq.git#af8cad8
  node         (core)
  ohmyposh     https://github.com/splinter98/asdf-ohmyposh#54b4f55
  p4           https://github.com/splinter98/asdf-p4.git#90482fd
  pacaptr      https://github.com/splinter98/asdf-pacaptr#31c67b9
  python       https://github.com/rtx-plugins/rtx-python.git#2b29b16
  ruby         (core)
  shellcheck   https://github.com/luizm/asdf-shellcheck.git#b160a52
  shfmt        https://github.com/luizm/asdf-shfmt.git#a42c5ff

toolset:
  python@latest, pacaptr@latest, p4@latest, cmake@latest, github-cli@latest, chezmoi@latest, bitwarden@latest, ohmyposh@latest

No problems found
```

**Additional context**
Fundamentally this is the same issue as #324 but isn't fixed by the implemented solution (avoid using https).
However, the proposed solution to switch to activating the reqweast feature `rustls-tls-native-roots` does fix the issue.

**Comment by inventfossil** [comment]
> @jdx I still have this issue even though I have the full certificate chain of the proxy available in Keychain on macOS 15.2:
> 
> ```
> 
> ╰─○ mise self-update --verbose
> 
> DEBUG ARGS: /Users/xxx/.local/bin/mise self-update --verbose
> 
> DEBUG starting new connection: https://api.github.com/
> 
> DEBUG proxy(http://127.0.0.1:3128) intercepts 'https://api.github.com/'
> 
> DEBUG starting new connection: https://api.github.com/
> 
> DEBUG proxy(http://127.0.0.1:3128) intercepts 'https://api.github.com/'
> 
> DEBUG starting new connection: https://api.github.com/
> 
> DEBUG proxy(http://127.0.0.1:3128) intercepts 'https://api.github.com/'
> 
> Checking target-arch... mise-v2025.1.6-macos-arm64.tar.gz
> 
> Checking current version... v2025.1.1
> 
> Checking latest released version... DEBUG starting new connection: https://api.github.com/
> 
> DEBUG proxy(http://127.0.0.1:3128) intercepts 'https://api.github.com/'
> 
> v2025.1.6 (5 versions compatible)
> 
> New release found! v2025.1.1 --> v2025.1.6
> 
> New release is compatible
> 
> 
> 
> mise release status:
> 
>   * Current exe: "/Users/xxx/.local/bin/mise"
> 
>   * New exe release: "mise-v2025.1.6-macos-arm64.tar.gz"
> 
>   * New exe download url: "https://api.github.com/repos/jdx/mise/releases/assets/219839442"
> 
> 
> 
> The new release will be downloaded/extracted and the existing binary will be replaced.
> 
> Do you want to continue? [Y/n] y
> 
> Downloading...
> 
> DEBUG starting new connection: https://api.github.com/
> 
> DEBUG proxy(http://127.0.0.1:3128) intercepts 'https://api.github.com/'
> 
> DEBUG redirecting 'https://api.github.com/repos/jdx/mise/releases/assets/219839442' to 'https://objects.githubusercontent.com/github-production-release-asset-2e65be/586920414/a73957ad-98d8-4687-8c6f-bdf39020f174?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250114%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250114T084743Z&X-Amz-Expires=300&X-Amz-Signature=f9ef44e9c96a1e455e05d9bc2952fec85d15c64f30868488868f454bcc69052a&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dmise-v2025.1.6-macos-arm64.tar.gz&response-content-type=application%2Foctet-stream'
> 
> DEBUG starting new connection: https://objects.githubusercontent.com/
> 
> DEBUG proxy(http://127.0.0.1:3128) intercepts 'https://objects.githubusercontent.com/'
> 
> Error:
> 
>    0: ReqwestError: error sending request for url (https://api.github.com/repos/jdx/mise/releases/assets/219839442)
> 
>    1: error sending request for url (https://api.github.com/repos/jdx/mise/releases/assets/219839442)
> 
>    2: client error (Connect)
> 
>    3: invalid peer certificate: UnknownIssuer
> 
> 
> 
> Location:
> 
>    src/cli/self_update.rs:99
> 
> 
> 
> Version:
> 
>    2025.1.1 macos-arm64 (dba7044 2025-01-06)
> 
> 
> 
> Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
> 
> Run with RUST_BACKTRACE=full to include source snippets.
> 
> ```

I have the exact same issue. Were you able to find a fix?

@jdx

**Comment by shinebayar-g** [comment]
Getting the same error. I don't have HTTPS_PROXY. But the company has installed custom certificate for github.com

```
❯ mise self-update --verbose
DEBUG ARGS: /Users/me/.local/bin/mise self-update --verbose
DEBUG starting new connection: https://api.github.com/
Checking target-arch... mise-v2025.6.1-macos-arm64.tar.gz
Checking current version... v2025.6.0
Checking latest released version... DEBUG starting new connection: https://api.github.com/
v2025.6.1 (1 versions compatible)
New release found! v2025.6.0 --> v2025.6.1
New release is compatible

mise release status:
  * Current exe: "/Users/me/.local/bin/mise"
  * New exe release: "mise-v2025.6.1-macos-arm64.tar.gz"
  * New exe download url: "https://api.github.com/repos/jdx/mise/releases/assets/262328799"

The new release will be downloaded/extracted and the existing binary will be replaced.
Do you want to continue? [Y/n]
Downloading...
DEBUG starting new connection: https://api.github.com/
DEBUG redirecting 'https://api.github.com/repos/jdx/mise/releases/assets/262328799' to 'https://objects.githubusercontent.com/github-production-release-asset-2e65be/586920414/75a6bab1-5694-4a29-87c9-6d650d630883?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250609%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250609T165719Z&X-Amz-Expires=300&X-Amz-Signature=48f3308d3b6e0e2b7a5ef0adcc9774dba824569fba5248b0a406c3d6ad642ca5&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dmise-v2025.6.1-macos-arm64.tar.gz&response-content-type=application%2Foctet-stream'
DEBUG starting new connection: https://objects.githubusercontent.com/
Error:
   0: ReqwestError: error sending request for url (https://api.github.com/repos/jdx/mise/releases/assets/262328799)
   1: error sending request for url (https://api.github.com/repos/jdx/mise/releases/assets/262328799)
   2: client error (Connect)
   3: invalid peer certificate: UnknownIssuer

Location:
   src/cli/self_update.rs:90

Version:
   2025.6.0 macos-arm64 (2025-06-02)

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```
